### PR TITLE
ILIAS7: UI Components, fix of responsive Logo Issue, see: #32016

### PR DESCRIPTION
**Pull request by thibsy**
Currently, the responsive Header Logo is not displayed on small screen sizes, the standard one is. See: https://mantis.ilias.de/view.php?id=32016

This PR is based on #4029 introduced by @Amstutz with the addition of a new `ReplaceResponsiveLogoCommand`. The command will replace the ILIAS default responsive header icon in skins, which have a custom `HeaderIcon.svg` but still the default ILIAS `HeaderIconResponsive.svg`. This solves the issue of displaying the wrong or default responsive icon when the site enters a certain width, for ppl who haven't used the responsive-icons yet.

### CODE REVIEW

File: src/Setup/CLI/ReplaceResponsiveLogoCommand.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            return null;
+        }
+
+        if (false !== ($hash = sha1_file($absolute_file_path))) {
+            return $hash;
+        }
+
+        return null;
+    }
+}
```
Comment (mjansenDatabay) [inline src/Setup/CLI/ReplaceResponsiveLogoCommand.php]: A line break is missing here ;-)

File: src/UI/Component/Layout/Page/Standard.php [hunk 2]
```diff
@@ -42,6 +42,13 @@ public function withMainbar(MainBar $main_bar) : Standard;
      */
     public function withLogo(Image $logo) : Standard;
 
+    /**
+     * @param Image $logo
+     *
```
Comment (mjansenDatabay) [inline src/UI/Component/Layout/Page/Standard.php]: Do the docs here (and below) provide any added value?

File: src/UI/Component/Layout/Page/Standard.php [hunk 3]
```diff
@@ -77,6 +89,11 @@ public function getBreadcrumbs();
      */
     public function getLogo();
 
+    /**
+     * @return Image|null
+     */
+    public function getResponsiveLogo();
```
Comment (mjansenDatabay) [inline src/UI/Component/Layout/Page/Standard.php]:
```suggestion
    public function getResponsiveLogo() : ?Image;
```

File: src/UI/Implementation/Component/Layout/Page/Standard.php [hunk 4]
```diff
@@ -225,6 +247,14 @@ public function getLogo()
         return $this->logo;
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function getResponsiveLogo()
```
Comment (mjansenDatabay) [inline src/UI/Implementation/Component/Layout/Page/Standard.php]:
```suggestion
    public function getResponsiveLogo() : ?Image
```

File: tests/UI/Component/Layout/Page/StandardPageTest.php [hunk 5]
```diff
@@ -122,6 +124,24 @@ public function testGetLogo()
         );
     }
 
+    public function testHasLogo()
```
Comment (mjansenDatabay) [inline tests/UI/Component/Layout/Page/StandardPageTest.php]:
```suggestion
    public function testHasLogo() : void
```

File: tests/UI/Component/Layout/Page/StandardPageTest.php [hunk 6]
```diff
@@ -122,6 +124,24 @@ public function testGetLogo()
...
+    {
+        $this->assertTrue($this->stdpage->hasLogo());
+    }
+
+    public function testGetResponsiveLogo()
```
Comment (mjansenDatabay) [inline tests/UI/Component/Layout/Page/StandardPageTest.php]:
```suggestion
    public function testGetResponsiveLogo() : void
```

File: tests/UI/Component/Layout/Page/StandardPageTest.php [hunk 7]
```diff
@@ -122,6 +124,24 @@ public function testGetLogo()
...
+    {
+        $this->assertEquals(
+            $this->responsive_logo,
+            $this->stdpage->getResponsiveLogo()
+        );
+    }
+
+    public function testHasResponsiveLogo()
```
Comment (mjansenDatabay) [inline tests/UI/Component/Layout/Page/StandardPageTest.php]:
```suggestion
    public function testHasResponsiveLogo() : void
```

File: src/Setup/CLI/ReplaceResponsiveLogoCommand.php [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+class ReplaceResponsiveLogoCommand extends Command
+{
+    protected const RESPONSIVE_LOGO_PATH = '/images/HeaderIconResponsive.svg';
+    protected const COMMON_LOGO_PATH = '/images/HeaderIcon.svg';
+
+    protected static $defaultName = 'replace-responsive-logo';
+    protected ?string $delos_responsive_logo_hash;
+    protected ?string $delos_common_logo_hash;
+    protected string $delos_path;
+    protected string $ilias_path;
```
Comment (mjansenDatabay) [inline src/Setup/CLI/ReplaceResponsiveLogoCommand.php]: Typed properties are not supported in PHP 7.3, which we must support in ILIAS 7.

Comment (thibsy) [reply src/Setup/CLI/ReplaceResponsiveLogoCommand.php]: Thanks for the review!

**Review by Amstutz — APPROVED** [review]
Hi @thibsy 

Thx a lot for the contribution! 

I am fine with the UI Changes. However, there seem to be many unrelated changes in here. Lots of strict typing here. That is all fine for trunk, however, in a productive release I am somewhat hesitant to introduced them as an unrelated change with a PR. IMO, we can make an exception here, since this will go to trunk as well. Take care with picking that to 6. I am not sure, if all those strict types will work as expected for 6. Who will check the CLI part @klees ?

Thx, 

@thibsy

**Comment by thibsy** [comment]
I noticed that adding the responsige-logo to the page constructor will be a breaking change for plugins that implement their own `PagePartProvider`. Therefore I undid that change for ILIAS 6&7 PR.

**Pull request merged**